typescript/src/api-doc-types/ApiDoc.ts

```diff
@@ -38,12 +38,28 @@ export abstract class BaseApiDoc implements ApiDoc {
 
   originalModule = this.fileInfo.projectRelativePath
     .replace(new RegExp("\." + this.fileInfo.extension + "$"), "");
-  typeChecker: TypeChecker = this.moduleDoc.typeChecker;
+  typeChecker: ts.TypeChecker = this.moduleDoc.typeChecker;
 
   constructor(public host: Host,
               public moduleDoc: ModuleDoc,
-              public symbol: Symbol,
-              public declaration: Declaration,
-              public aliasSymbol?: Symbol) {
+              public symbol: ts.Symbol,
+              public declaration: ts.Declaration,
+              public aliasSymbol?: ts.Symbol) {
+  }
+
+  protected getTypeString(decl: ts.Declaration): string | undefined {
+    if (!ts.isVariableDeclaration(decl) && !ts.isPropertyDeclaration(decl) && !ts.isParameter(decl)) {
+      return undefined;
+    }
+
+    if (!decl.type && !decl.initializer) {
+      return undefined;
+    }
+
+    const type = decl.type ?
+      this.typeChecker.getTypeFromTypeNode(decl.type) :
+      this.typeChecker.getTypeAtLocation(decl);
```

Comment:
Note that we want to get the type at the declaration, not the initializer, otherwise we end up with the type being too constrained. E.g. `"some-string"` rather than `string`.

Reply:
And I assume in case there is no explicit type, the `getTypeAtLocation` call returns the type of "declaration" but not _explicitly_ restricted to the initializer value?

If so, would it be worth adding this as a comment?

Reply:
By looking at it in Material, for something like: 

```ts
/** The total height of the autocomplete panel. */
export const AUTOCOMPLETE_PANEL_HEIGHT = 256;
```

it still returns `256` as type while it should be technically `number` based on the expected behavior?

Reply:
Actually because it is a const it's type really is 256 :-(. If it was a `let` the it would be a number.

Reply:
Ah damn, hahah 😄 I oversaw that it's a const. Makes sense since you cannot change the variable 🤦‍♂️

Reply:
I think it should be good then. Works fine on Material.

Reply:
> And I assume in case there is no explicit type, the getTypeAtLocation call returns the type of "declaration" but not explicitly restricted to the initializer value?

Exactly!